### PR TITLE
Feature/block screenshots

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

The only thing left is to add the methods where we want to block screenshots

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #

### CODE REVIEW

File: app/components/Views/ImportPrivateKeySuccess/index.js [hunk 1]
```diff
@@ -90,15 +100,17 @@ class ImportPrivateKeySuccess extends PureComponent {
 			Logger.error(e, 'Error while refreshing imported pkey');
 		}
 		InteractionManager.runAfterInteractions(() => {
+			PreventScreenshot.forbid();
 			BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
 		});
 	};
 
-	componentWillUnmount() {
+	componentWillUnmount = async () => {
```
Comment (andrepimenta) [inline app/components/Views/ImportPrivateKeySuccess/index.js:108]: `async` should not be necessary?

File: app/components/Views/Wallet/index.js [hunk 2]
```diff
@@ -106,11 +106,11 @@ class Wallet extends PureComponent {
 		this.mounted = true;
 	}
 
-	componentDidMount() {
+	componentDidMount = async () => {
```
Comment (andrepimenta) [inline app/components/Views/Wallet/index.js:109]: `async` should not be necessary?

File: app/components/Views/RevealPrivateCredential/index.js [hunk 3]
```diff
@@ -169,8 +181,17 @@ class RevealPrivateCredential extends PureComponent {
 				}
 			}
 		}
+		InteractionManager.runAfterInteractions(() => {
+			PreventScreenshot.forbid();
+		});
 	}
 
+	componentWillUnmount = async () => {
```
Comment (andrepimenta) [inline app/components/Views/RevealPrivateCredential/index.js:189]: `async` should not be necessary?

**Review by andrepimenta — CHANGES_REQUESTED** [review]
If this is not implemented on iOS, then the respective NativeModule is not available right? This should be checked, otherwise will crash on iOS.

Also, it seems that there is no prevention for other apps to record screen, for example. Is it not possible?

Otherwise, looking good! 👍

**Comment by estebanmino** [comment]
@andrepimenta  good call on iOS 👌

**Comment by estebanmino** [comment]
@andrepimenta all good now, added a check for iOS. Regarding the screen recording, I added the `FLAG_SECURE` flag that blocks screenshots and recording ([related link](https://developer.android.com/reference/android/view/Display?authuser=2#FLAG_SECURE))

**Comment by ibrahimtaveras00** [comment]
**This works on Android on the following views:**
- import seed phrase
- security & privacy > "SHOW PRIVATE KEY"
- security & privacy > "REVEAL SEED PHRASE"

showing this message on my pixel 2 
image

**Few places where this doesn't work:**
- on the "secure your wallet" flow after creating your password and then navigating to the "Your seed phrase" view 
- on the import account view when adding the private key

**Comment by andrepimenta** [comment]
@estebanmino We should consider creating a service or helper called, for example, `PreventScreenshot` that implements empty functions for iOS, so that we don't have to duplicate code like this:
`const PreventScreenshot = Platform.OS === 'android' ? NativeModules.PreventScreenshot : null;` and `PreventScreenshot && PreventScreenshot.forbid();`

A possible implementation could be
PreventScreenshot.js
`
import { NativeModules } from react-native

const isAndroid = Platform.OS === 'android' 
export default {
forbid: isAndroid ? () => NativeModules.PreventScreenshot.forbid() : () => {},
allow: isAndroid ? () => NativeModules.PreventScreenshot.allow() : () => {}
}
`

What do you think?

**Comment by estebanmino** [comment]
@andrepimenta @ibrahimtaveras00 issues addressed 👌

**Review by andrepimenta — APPROVED** [review]
Great! Looks good to me 👌

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good on Android, QA Passed 👍

**Comment by omnat** [comment]
In the latest, on iOS, I could still take screenshots on these flows:
- reveal seedphrase
- during the backup flow

From my understanding, we can not prevent the user on iOS on taking a screenshot (unlike on Android), but we can show them a "taking a screenshot is not safe to do.."

**Pull request merged**